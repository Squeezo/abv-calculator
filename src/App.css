.app {
  display: flex;
  align-content: center;
  justify-content: center;
  padding:20px;
}

.header {
  background-color: #282c34;
  display: flex;
  align-content: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

fieldset {
  padding: 20px;
  width:30vw;
  font-size:1.3em;
  border-radius: 5px;
}

label {
  padding-right:10px;
}
legend {
  width:100%;
  text-align: center;
  background-color: #ccc;
  padding:4px;
  border-radius: 5px;
}

.form {
  display: flex;
  justify-content: space-between;
}

.fields {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  
}

.submit {
  display: flex;
  justify-content: flex-end;
}

.abv-display {
  text-align: left;
  margin: 0;
}

.abv {
  font-size:2em;
}
input[type=number] {
  width:60px;
  padding:4px;
  font-size:1.1rem;
  margin:10px 0;
}

button {
  width:10vw;
  height:44px;
}