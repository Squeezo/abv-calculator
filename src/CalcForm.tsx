import * as React from 'react';

interface State {
  og: number;
  fg: number;
  message: string;
};

export default class CalcForm extends React.Component<{}, State> {
  state: State = {
    og: 1.055,
    fg: 1.015,
    message: ''
  };

  validate = (value: number) => {
    if(!value) {
      return false;
    }

    if(value < 1) {
      this.setState(Object.assign({}, this.state, { message: "Gravity must be greater than 0.999" } ));
      return false;
    }

    if(value > 1.999) {
      this.setState(Object.assign({}, this.state, { message: "Gravity must be 1.999 or less" } ));
      return false;
    }

    return true;

  }

  handleChange = (e: React.FormEvent<HTMLInputElement>) => {
      
      let name= e.currentTarget.name;
      let value = Number.parseFloat(e.currentTarget.value);

      let valid = this.validate(value)

      if(valid) {
        this.setState(Object.assign({}, this.state, { [name]: value, message: "" } ));
      }
  };

  render () {

    return (
      <fieldset>
        <legend>Enter your gravity values below:</legend>
        <div className='form'>
          <div>
              <label htmlFor='og'>OG:</label>
              <input name='og' id='og' type='number' step={0.001} min={1.00} max={1.999} value={this.state.og} onChange={this.handleChange} />
          </div>
          <div>
              <label htmlFor='fg'>FG: </label><input name='fg' id='fg' type='number' step={0.001} value={this.state.fg} onChange={this.handleChange} />
          </div>
          <div className='abv-display'>
            <span className='label'>ABV: </span>
            <span className='abv'>{GetAbv({ og: this.state.og, fg: this.state.fg })}%</span>
          </div>           
        </div>


      </fieldset>
    );
  }
}

export const GetAbv = ({ og, fg }: { og: number; fg: number; }) : any => {
  if (!og || !fg) return 0;
  if((og && typeof(og) === 'string') || (fg && typeof(fg) === 'string')) return 0;
  if((og && og < 1) || (fg && fg < 1)) return 0;
  if((og && og > 1.999) || (fg && fg > 1.999)) return 0;
  return Math.floor(((og - fg) * 131.25) * 100) / 100;
}