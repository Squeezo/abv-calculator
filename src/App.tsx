import React from 'react';
import CalcForm from './CalcForm';
import './App.css';

const App: React.FC = () => {
  return (
    <div className='app'>
      <CalcForm />
    </div>
  );
}

export default App;
