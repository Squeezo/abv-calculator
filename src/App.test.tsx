import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme'; 
import CalcForm from './CalcForm';

describe("App Component", ()=> {

  it('contains a div with a class of .app', ()=>{
    const app = shallow(<App />);
    expect(app.find('.app')).toHaveLength(1);
  });

  it('renders a single CalcForm component', ()=> {
    const app = shallow(<App />);
    expect(app.find(CalcForm)).toHaveLength(1);
  });

  it('renders as expected', ()=> {
    const app = shallow(<App />);
    expect(app).toMatchSnapshot();
  });
});