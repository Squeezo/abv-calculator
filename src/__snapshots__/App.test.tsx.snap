// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders as expected 1`] = `ShallowWrapper {}`;
